### PR TITLE
[FIX] base_rest_pydantic: forword service to PydanticModel.from_params 

### PR DESCRIPTION
**Pull request by petrus-v**
...while managing PydanticModelList.from_params 

To reproduce create a service use PydanticModelList in input param and call that service

### CODE REVIEW

**Comment by petrus-v** [comment]
@lmignon I would be happy to get reviews from yon on this pydantic topic !

File: base_rest_pydantic/tests/test_from_params.py [hunk 1]
```diff
@@ -22,8 +22,14 @@ class Model1(BaseModel):
 
         self.Model1: BaseModel = Model1
 
-    def _from_params(self, pydantic_cls: Type[BaseModel], params: dict, **kwargs):
-        restapi_pydantic = restapi.PydanticModel(pydantic_cls, **kwargs)
+    def _from_params(
+        self,
+        pydantic_cls: Type[BaseModel],
```
Comment (lmignon) [inline base_rest_pydantic/tests/test_from_params.py]: @petrus-v Something is wrong here. The second arg is the service not a BaseModel type.

**Review by lmignon — CHANGES_REQUESTED** [review]

File: base_rest_pydantic/tests/test_from_params.py (hunk 1 above)
Comment (petrus-v) [reply base_rest_pydantic/tests/test_from_params.py]: I don't get it :-/... may be we should rename this method in unit test that call the from_params method on the PydanticModel for the given BaseModel. 

Service is moked bellow before calling from_params !

Do I missed something ?

Comment (petrus-v) [reply base_rest_pydantic/tests/test_from_params.py]: we could call it `_from_params_call` ?

Comment (lmignon) [reply base_rest_pydantic/tests/test_from_params.py]: @petrus-v My bad, we are into the tests module....

Why do you need to pass the restapi class to the `_from_params' method. We don't need it. IMO we've 2 possibilities:
* `restapi_pydantic = restapi.PydanticModelList if isinstance(params, list) else restapi.PydanticModel`
* Add an explicit parameter to the method to specify that the params are a list...

Comment (petrus-v) [reply base_rest_pydantic/tests/test_from_params.py]: I'm fine with your first suggestion, providing the restapi class as parameter was a way for me to explicitly test what I wanted to test here !

I'll change that !

File: base_rest_pydantic/tests/test_from_params.py [hunk 2]
```diff
@@ -44,3 +49,15 @@ def test_from_params_missing_required_field(self):
...
         with self.assertRaisesRegex(UserError, msg):
             self._from_params(self.Model1, {"description": "Instance Description"})
+
+    def test_from_params_pydantic_model_list(self):
+        params = [
+            {"name": "Instance Name", "description": "Instance Description"},
+            {"name": "Instance Name 2", "description": "Instance Description 2"},
+        ]
+        instances = self._from_params(self.Model1, params)
+        restapi.PydanticModelList(self.Model1)
```
Comment (lmignon) [inline base_rest_pydantic/tests/test_from_params.py]: this line is useless

Comment (petrus-v) [reply base_rest_pydantic/tests/test_from_params.py]: sure :sweat_smile:

Comment (petrus-v) [reply base_rest_pydantic/tests/test_from_params.py]: here you are !

**Review by lmignon — APPROVED** [review]
Thank you @petrus-v

**Review by PierrickBrun — APPROVED** [review]
Thanks

**Pull request merged**